sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -221,7 +221,7 @@ object FileFormatWriter extends Logging {
       val (_, duration) = Utils.timeTakenMs { committer.commitJob(job, commitMsgs) }
       logInfo(s"Write Job ${description.uuid} committed. Elapsed time: $duration ms.")
 
-      processStats(description.statsTrackers, ret.map(_.summary.stats))
+      processStats(description.statsTrackers, ret.map(_.summary.stats), duration)
```

Comment:
Sorry but why do we need the show the duration of the function call of `commitJob` here.
As per the doc:
```
  /**
   * Commits a job after the writes succeed. Must be called on the driver.
   */
  def commitJob(jobContext: JobContext, taskCommits: Seq[TaskCommitMessage]): Unit
```
The commitJob API mostly is for moving the temporary output files to the target final path.

Reply:
Some times  after task all completed, we wait a long time then job finished, it's always cost on job commit and  metadata handling such as `externalCatalog.loadDynamicPartition()` etc.
 
These duration information is important when it's slow  when we want to compare job's performance. Since when hdfs is unstable, file operation will cost  more time. 

Also I want to add metrics about metadata handling time after job committed.

Reply:
Normally, this should be fast. I don't think it is a good idea to put this in the SQL graph.
Besides, the name "duration of committing the job" can be confusing to end-users.
I have to leave -1 for this one.

Reply:
`externalCatalog.loadDynamicPartition()` is this really counted as the commit duration in this PR?

Reply:
> `externalCatalog.loadDynamicPartition()` is this really counted as the commit duration in this PR?

Not yet,  we need to collect this duration after job committed. not counted in job commit duration.

Reply:
Just now, my friend ask me why job finished then cost 80s to job committed.
```
21/02/25 15:42:12 INFO DAGScheduler: ResultStage 1 (run at AccessController.java:0) finished in 82.189 s
21/02/25 15:42:12 INFO DAGScheduler: Job 1 finished: run at AccessController.java:0, took 84.330846 s
21/02/25 15:43:38 INFO FileFormatWriter: Job null committed.
21/02/25 15:43:38 WARN DFSClient: Slow ReadProcessor read fields took 41202ms (threshold=30000ms); ack: seqno: 140 status: SUCCESS downstreamAckTimeNanos: 33201980 4: "\000", targets: [172.16.1.71:9866, 172.16.1.104:9866, 172.16.1.18:9866, 172.16.1.33:9866]
```

His SQL task run 80s, job commit cost 80s and hive metadata load data cost 100s.

Reply:
if we show job commit in the UI, shall we also show hive load table?

Reply:
> if we show job commit duration in the UI, shall we also show hive load table duration?

I originally planned to show hive load table duration after this PR. Shall I also update about hive load table duration in this PR?

Reply:
https://github.com/apache/spark/pull/31679 this is a related PR before add hive load table duration in SQLMetrics, since `CreateTableAsSelect` also need show hive load data duration.

I hope to add hive load data duration after this pr and https://github.com/apache/spark/pull/31679.

Reply:
Gentle ping @cloud-fan What should I do next for this pr?